runtime/wasmer.go

```diff
@@ -157,6 +159,11 @@ func ext_get_allocated_storage(context unsafe.Pointer, keyData, keyLen, writtenO
 		return 0
 	}
 
+	if val == nil {
+		memory[writtenOut] = 0
```

Comment:
The spec says `memory[writtenOut]` should return 2^32-1 if the entry does not exist.

Reply:
Not sure if you had a conversation with Skalman that indicates we should do otherwise though.

Reply:
Went through the rest of the PR, everything else looks good! Let me know re:above :)

Reply:
after looking at the rust code, it looks like they write to memory at writtenOut to writtenOut+4. I have updated to code accordingly